content/docs/api-reference/dvcfs.md

```diff
@@ -0,0 +1,66 @@
+# DvcFileSystem
+
+DvcFileSystem provides a pythonic file interface (aka
+[fsspec](https://filesystem-spec.readthedocs.io/)) for a DVC repo and provides a
+read-only filesystem-like operations like `ls`, `du`, `glob`, `get`, `download`,
+etc.
+
+DvcFileSystem provides a single view of all the files/directories in your
+repository, be it Git-tracked or DVC-tracked, or untracked (in case of a local
+repository). DvcFileSystem is smart to reuse the files in your cache directory
```

Comment:
Internally it avoids downloading DVC-tracked files unless it's needed. It's using DVC cache [make it abbr] to avoid downloading objects every time and support streaming for supported remote storages [link] for operations like ...